tcpclient/connectionsManagement.go

```diff
@@ -1,30 +1,46 @@
 package tcpclient
 
 import (
-	"sync"
+	"bufio"
 	"fmt"
+	"io"
 	"net"
-	"bufio"
 	"strconv"
+	"sync"
 )
 
-func TcpConnect(id int, host string, port int, wg *sync.WaitGroup) error {
-	fmt.Println("\t runner " + strconv.Itoa(id) + " is initiating a connection")
+func reportConnectionStatus(debugOut io.Writer, statusChannel chan<- Connection, connectionDescription Connection)  {
+	statusChannel <- connectionDescription
+	fmt.Fprintln(debugOut, "\t", connectionDescription)
+}
+
+func TcpConnect(id int, host string, port int, wg *sync.WaitGroup, debugOut io.Writer, statusChannel chan<- Connection) error {
+	connectionDescription := Connection{
+		Id: id,
+		status: ConnectionDialing,
+	}
+	reportConnectionStatus(debugOut, statusChannel, connectionDescription)
 	conn, err := net.Dial("tcp", host+":"+strconv.Itoa(port))
 	if err != nil {
-		fmt.Println(err)
+		connectionDescription.status = ConnectionError
+		reportConnectionStatus(debugOut, statusChannel, connectionDescription)
+		fmt.Fprintln(debugOut, "Connection", id, "was unable to open the connection. Error:")
+		fmt.Fprintln(debugOut, err)
 		wg.Done()
 		return err
 	}
-	fmt.Println("\t runner " + strconv.Itoa(id) + " established the connection")
+
+	connectionDescription.status = ConnectionEstablished
+	reportConnectionStatus(debugOut, statusChannel, connectionDescription)
 	connBuf := bufio.NewReader(conn)
 	for {
 		str, err := connBuf.ReadString('\n')
 		if len(str) > 0 {
-			fmt.Println(str)
+			fmt.Fprintln(debugOut, "Connection", id, "got", str)
 		}
 		if err != nil {
-			fmt.Println("\t runner " + strconv.Itoa(id) + " got its connection closed")
+			connectionDescription.status = ConnectionClosed
+			reportConnectionStatus(debugOut, statusChannel, connectionDescription)
```

Comment:
shall we add a report message here? (like `error`)

Reply:
I don't think so... in fact, all connections will eventually become closed in the current logic (so reaching this point is not actually an error: its just how the goroutine will finish..)

Reply:
:+1